docs/saved_objects/saved_object_access_control_design_proposal.md

```diff
@@ -0,0 +1,216 @@
+
+OpenSearch Dashboards uses the [saved objects APIs](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/core/server/saved_objects/routes) to perform CRUD operations against saved objects. And in order to apply the access control to the OpenSearch Dashboards data, we can implement a privilege evaluator component and embed it into the saved object database operations to enforce that users can only operate the data they have access to.
+
+For single and bulk CRUD operations, the operations are performed by saved object ids, thus OpenSearch Dashboards can fetch the saved object ACLs by the saved object id and evaluate the user permission before sending the response back to client side. For common saved objects, the privilege evaluator will evaluate the ACLs of the saved object itself and the ACLs of the workspaces that object is associated to to determine if that operation should be allowed or denied. While for workspaces, it only needs to evaluate the ACL of that workspace itself.
+
+For the [find](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/routes/find.ts) API which lists the saved objects based on different search conditions, we will need to construct a search filter that can be applied to the query when performing searches against database.
+
+OpenSearch Dashboards provided a couple approaches to intercept saved objects operations, such as using [ClientWrapper](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/service/lib/scoped_client_provider.ts#L74) and [SavedObjectsClientFactory](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/service/lib/scoped_client_provider.ts#L82) . We can leverage these interception mechanisms to simplify the privilege evaluation implementation.
+
+
+#### Where to store ACLs
+
+The ACL is a logically a sub-resource/attribute that is attached to OpenSearch Dashboards resources, and the same access control mechanism should apply to the ACL as the saved object it attaches to.  We can either embed the ACL as attribute to the saved object, or store it separately as the standalone record, depending on what database is used by OpenSearch Dashboards. e.g. in the case of using NoSQL database as OpenSearch Dashboards metadata store, such as using OpenSearch index, embed the ACL into the saved object document as an attribute can simplify the privilege evaluation. while if using relational DB as OpenSearch Dashboards metadata store, storing ACLs as separate data records will be simpler.
```

Comment:
If a user can 'write' to an object, it seems like they can alter its ACLs, this seems like a way to circumvent the permissions?  Seems like the ACL associate needs to be protected from this level of access - otherwise anyone with a write allowed connection string is effectively a workspace super admin